<?xml version="1.0"?>
<launch>
    <arg name="ns"                                  default="" />

    <arg name="template_file_dir"                   default="vitamin/ptcld_9.pcd">
    <arg name="template_pointcloud_filename"        default="$(find arm_vs)/dataset/$(arg template_file_dir)" />
    
    <arg name="input_pointcloud"                    default="$(arg ns)/voxel_grid/output" />
    <arg name="tracked_pointcloud"                  default="$(arg ns)/tracked_pointcloud" />
    <arg name="output_pointcloud"                   default="$(arg ns)/aligned_pointcloud" />

    <arg name="tracked_template_frame"              default="particle_filter_tracked_template_frame" />
    <arg name="tracker_reset_service"               default="reset_particle_filter_tracker" />

    <arg name="icp_alignment_frame"                 default="icp_alignment_frame" />

    <node pkg="arm_vs" type="arm_vs_track_obj_pointcloud" name="pointcloud_tracking" output="screen">
        <param name="input_pointcloud_topic" type="string" value="$(arg input_pointcloud)" />
        <param name="output_pointcloud_topic" type="string" value="$(arg tracked_pointcloud)" />
        <param name="template_cloud_filename" type="string" value="$(arg template_pointcloud_filename)" />
        <param name="particle_filter_tracked_template_frame" type="string" value="$(arg tracked_template_frame)" />
        <param name="particle_filter_tracker_reset_service" type="string" value="$(arg tracker_reset_service)" />
    </node>

    <node pkg="arm_vs" type="arm_vs_align_obj_pointcloud" name="pointcloud_alignment" output="screen">
        <param name="input_template_pointcloud_topic" type="string" value="$(arg tracked_pointcloud)" />
        <param name="input_scene_pointcloud_topic" type="string" value="$(arg input_pointcloud)" />
        <param name="output_pointcloud_topic" type="string" value="$(arg output_pointcloud)" />
        <param name="icp_alignment_frame" type="string" value="$(arg icp_alignment_frame)" />
    </node>
</launch>